modules/speech-to-text-impl/src/main/java/org/opencastproject/speechtotext/impl/engine/WhisperEngine.java

```diff
@@ -143,56 +150,63 @@ public Map<String, Object> generateSubtitlesFile(File mediaFile,
         "--model", whisperModel,
         "--output_dir", preparedOutputFile.getParent()};
 
-    List<String> command = new ArrayList<>(Arrays.asList(baseCommands));
+    List<String> transcriptionCommand = new ArrayList<>(Arrays.asList(baseCommands));
 
     if (translate) {
-      command.add("--task");
-      command.add("translate");
+      transcriptionCommand.add("--task");
+      transcriptionCommand.add("translate");
       logger.debug("Translation enabled");
       language = "en";
     }
 
     if (!language.isBlank() && !translate) {
       logger.debug("Using language {} from workflows", language);
-      command.add("--language");
-      command.add(language);
+      transcriptionCommand.add("--language");
+      transcriptionCommand.add(language);
     }
 
     if (quantization.isSome()) {
       logger.debug("Using quantization {}", quantization.get());
-      command.add("--compute_type");
-      command.add(quantization.get().toString());
+      transcriptionCommand.add("--compute_type");
+      transcriptionCommand.add(quantization.get().toString());
     }
 
     if (isVADEnabled.isSome()) {
       logger.debug("Setting VAD to {}", isVADEnabled.get());
-      command.add("--vad_filter");
-      command.add(isVADEnabled.get().toString());
+      transcriptionCommand.add("--vad_filter");
+      transcriptionCommand.add(isVADEnabled.get().toString());
     }
 
-    logger.info("Executing Whisper's transcription command: {}", command);
-
-    Process process = null;
+    logger.info("Executing Whisper's transcription command: {}", transcriptionCommand);
 
+    Process transcriptonProcess = null;
+    BufferedReader in = null;
     String mediaFileNameWithoutExtension;
 
     try {
-      ProcessBuilder processBuilder = new ProcessBuilder(command);
+      ProcessBuilder processBuilder = new ProcessBuilder(transcriptionCommand);
       processBuilder.redirectErrorStream(true);
-      process = processBuilder.start();
+      transcriptonProcess = processBuilder.start();
 
+      // tell listeners about output
+      in = new BufferedReader(new InputStreamReader(transcriptonProcess.getInputStream()));
+      String line;
+      while ((line = in.readLine()) != null) {
+        handleTranscriptionOutput(line);
+      }
 
       // wait until the task is finished
-      int exitCode = process.waitFor();
+      int exitCode = transcriptonProcess.waitFor();
       logger.debug("Whisper process finished with exit code {}",exitCode);
 
       if (exitCode != 0) {
         var error = "";
-        try (var errorStream = process.getInputStream()) {
+        try (var errorStream = transcriptonProcess.getInputStream()) {
```

Comment:
This doesn't really make sense anymore. You already read all the input stream, didn't you? The stream should be empty here.

Reply:
I removed this part.